drivers/display/display_smartbond.c

```diff
@@ -0,0 +1,634 @@
+
+		data->dma_block_cfg.dest_address = (uint32_t)dst;
+		data->dma_block_cfg.source_address = (uint32_t)src;
+
+		if (dma_config(data->dma, data->dma_channel, &data->dma_cfg)) {
+			LOG_ERR("Could not configure DMA");
+			k_sem_give(&data->device_sem);
+			return -EIO;
+		}
+
+		if (dma_start(data->dma, data->dma_channel)) {
+			LOG_ERR("Could not start DMA");
+			k_sem_give(&data->device_sem);
+			return -EIO;
+		}
+
+		k_sem_take(&data->dma_sync_sem, K_FOREVER);
+
+		src += data->layer.stride;
+		dst += (desc->pitch * data->current_pixel_size);
+	}
+
+	if (dma_stop(data->dma, data->dma_channel)) {
+		LOG_WRN("Could not stop DMA");
+	}
+
+	k_sem_give(&data->device_sem);
+
+	return 0;
+}
+
+static int display_smartbond_write(const struct device *dev,
+				const uint16_t x, const uint16_t y,
+				const struct display_buffer_descriptor *desc,
+				const void *buf)
+{
+	struct display_smartbond_data *data = dev->data;
+	uint8_t *dst = data->buffer;
+	const uint8_t *src = buf;
+
+	k_sem_take(&data->device_sem, K_FOREVER);
+
+	/* pointer to upper left pixel of the rectangle */
+	dst += (x * data->current_pixel_size);
+	dst += (y * data->layer.stride);
+
+	/*
+	 * If continuous mode is enabled, wait for the current frame to finish.
+	 * Do not disable continuous mode as this will have visual artifacts.
+	 */
+	if (LCDC->LCDC_MODE_REG & LCDC_LCDC_MODE_REG_LCDC_MODE_EN_Msk) {
+		LCDC->LCDC_INTERRUPT_REG |= LCDC_LCDC_INTERRUPT_REG_LCDC_VSYNC_IRQ_EN_Msk;
+		k_sem_take(&data->sync_sem, K_FOREVER);
+	}
+
+	data->dma_block_cfg.block_size = desc->width * data->current_pixel_size;
+	/*
+	 * Source and destination base address is word aligned.
+	 * Data size should be selected based on color depth as
+	 * cursor is shifted multiple of pixel color depth.
+	 */
+	data->dma_cfg.source_data_size = data->dma_cfg.dest_data_size =
+			!(data->current_pixel_size & 3) ? 4 :
+			!(data->current_pixel_size & 1) ? 2 : 1;
+
+	data->dma_cfg.dest_burst_length = data->dma_cfg.source_burst_length =
+		!((data->dma_block_cfg.block_size / data->dma_cfg.source_data_size) & 7) ? 8 :
+		!((data->dma_block_cfg.block_size / data->dma_cfg.source_data_size) & 3) ? 4 : 1;
+
+	for (int row = 0; row < desc->height; row++) {
+
+		data->dma_block_cfg.dest_address = (uint32_t)dst;
+		data->dma_block_cfg.source_address = (uint32_t)src;
+
+		if (dma_config(data->dma, data->dma_channel, &data->dma_cfg)) {
+			LOG_ERR("Could not configure DMA");
+			k_sem_give(&data->device_sem);
+			return -EIO;
+		}
+
+		if (dma_start(data->dma, data->dma_channel)) {
+			LOG_ERR("Could not start DMA");
+			k_sem_give(&data->device_sem);
+			return -EIO;
+		}
+
+		k_sem_take(&data->dma_sync_sem, K_FOREVER);
+
+		dst += data->layer.stride;
+		src += (desc->pitch * data->current_pixel_size);
+	}
+
+	if (dma_stop(data->dma, data->dma_channel)) {
+		LOG_WRN("Could not stop DMA");
+	}
+
+	/* Make sure continus mode is enabled */
+	LCDC->LCDC_MODE_REG |= LCDC_LCDC_MODE_REG_LCDC_MODE_EN_Msk;
```

Comment:
Note- I'm not familiar with this IP, just curious- do we need to set this bit here? IE is there some condition that would cause the IP block to disable continuous mode while executing, and require us to force it back on? It seems like once we turn continuous mode on, it would stay on.

Also- spelling of comment above this line needs to be fixed

Reply:
As I mentioned below, this is a leftover from previous implementation. I will move this to the initialization routine.